src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -1194,7 +1220,7 @@ internal static BinderCheckTypeResult CheckUnaryOpCore(IErrorContainer errorCont
             switch (node.Op)
             {
                 case UnaryOp.Not:
-                    var resNot = CheckTypeCore(errorContainer, node.Child, features, childType, DType.Boolean, /* coerced: */ DType.Number, DType.Decimal, DType.String, DType.OptionSetValue, DType.UntypedObject);
```

Comment:
>DType.OptionSetValue [](http://example.com/codeflow?start=169&length=20)

Can't we do something like `Not MyBoolOptionSet.NoValue`?

Reply:
I believe you are asking about https://github.com/microsoft/Power-Fx/blob/0932cfac9d2735b87705db1b4e3d6749a0b06861/src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/StronglyTypedEnum_TestEnums.txt#L548 and tests near this one.  Instead of listing OptionSetValue as a coercable type, CheckTypeCore will coerce any option set with a matching backing type.  This simplifies the logic in that function, prevents indirect coercions (string backed option set to Boolean), and makes for a better error message for the maker.